Block/BrowseAbandonment.php

```diff
@@ -0,0 +1,293 @@
+     */
+    private $coreRegistry;
+
+    /**
+     * @var CustomerModelSession
+     */
+    private $customerModelSession;
+
+    /**
+     * @var CustomerModelSession
+     */
+    private $request;
+
+    /**
+     * @var CheckoutSessionFactory
+     */
+    private $checkoutSessionFactory;
+
+    /**
+     * BrowseAbandonment constructor.
+     * @param Context $context
+     * @param YotpoConfig $yotpoConfig
+     * @param Registry $coreRegistry
+     * @param CustomerModelSession $customerModelSession
+     * @param HttpRequest $request
+     * @param CheckoutSessionFactory $checkoutSessionFactory
+     * @param array<mixed> $data
+     */
+    public function __construct(
+        Context $context,
+        YotpoConfig $yotpoConfig,
+        Registry $coreRegistry,
+        CustomerModelSession $customerModelSession,
+        HttpRequest $request,
+        CheckoutSessionFactory $checkoutSessionFactory,
+        array $data = []
+    ) {
+        $this->yotpoConfig = $yotpoConfig;
+        $this->coreRegistry = $coreRegistry;
+        $this->customerModelSession = $customerModelSession;
+        $this->request = $request;
+        $this->checkoutSessionFactory = $checkoutSessionFactory;
+        parent::__construct($context, $data);
+    }
+
+    /**
+     * Check module is enabled and API tokens set
+     *
+     * @return boolean
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     */
+    public function isAppKeyAndSecretSet()
+    {
+        return $this->yotpoConfig->isAppKeyAndSecretSet();
+    }
+
+    /**
+     * Get if page type is eligible for Browse Abandonment event
+     *
+     * @return boolean
+     */
+    public function isPageTypeEligibleForBrowseAbandonmentEvent()
+    {
+        $magentoPageType = $this->getMagentoPageType();
+        return in_array($magentoPageType, self::ELIGIBLE_MAGENTO_PAGE_TYPES_FOR_BROWSE_ABANDONMENT);
+    }
+
+    /**
+     * Get Browse Abandonment event info
+     *
+     * @return string
+     */
+    public function getBrowseAbandonmentEventInfo()
+    {
+        $browseAbandonmentPageType = $this->getBrowseAbandonmentPageType();
+
+        $browseAbandonmentEventInfo = [];
+        $browseAbandonmentEventInfo['type'] = $browseAbandonmentPageType;
+        $browseAbandonmentEventInfo['data'] = $this->getBrowseAbandonmentEventInfoData($browseAbandonmentPageType);
+        return json_encode($browseAbandonmentEventInfo);
+    }
+
+    /**
+     * Get Browse Abandonment event info data
+     *
+     * @return array
+     */
+    private function getBrowseAbandonmentEventInfoData($browseAbandonmentPageType) {
+        $browseAbandonmentInfoData = [];
+        $browseAbandonmentInfoData['type'] = $browseAbandonmentPageType;
+        $browseAbandonmentInfoData['store_id'] = $this->getAppKey();
+
+        $idOfBrowseAbandonmentPageType = $this->getIdByBrowseAbandonmentPageType($browseAbandonmentPageType);
+        if ($idOfBrowseAbandonmentPageType) {
```

Comment:
I actually think your suggestion is more complicated.
Can you explain why the name I gave is tricky? :)

Reply:
because the `id` is not of the browse abandonment page type, it not the page type id, its more of an entity that is used in that page type, but uncoupled directly. Can checkout page contain customer id? it can right? but the page type is checkout, and so on, the id you get in return in indication of the entity used specially in that page type, on not of it.

Reply:
I agree. Thank you for the insights!